src/main/java/com/rhombus/api/examples/DoorReport.java

```diff
@@ -0,0 +1,211 @@
+            dateFormat.setTimeZone(TimeZone.getTimeZone("America/Los_Angeles"));
+            startTimeMilli = dateFormat.parse(startTime).getTime();
+        }
+        else {
+            // if no command line argument, startTime defaults to 24 hours ago (milliseconds)
+            startTimeMilli = ZonedDateTime.now().toInstant().toEpochMilli() - (1000 * 60 * 60 * 24);
+        }
+        if (commandLine.hasOption("endTime")) {
+            // converting string endTime argument to milliseconds
+            DateFormat dateFormat = new SimpleDateFormat("yyyy-mm-dd hh:mm:ss");
+            dateFormat.setTimeZone(TimeZone.getTimeZone("America/Los_Angeles"));
+            endTimeMilli = dateFormat.parse(endTime).getTime();
+        }
+        else {
+            // if no command line argument, endTime defaults to now (milliseconds)
+            endTimeMilli = ZonedDateTime.now().toInstant().toEpochMilli();
+        }
+
+        _initialize(apiKey);
+        String sensorUuid = uuidConvert(sensorName);
+        // input validation for the sensor name
+        if (sensorUuid.equals(" "))
+        {
+            System.out.println("Invalid sensor name");
+            System.exit(0);
+        }
+        File csvFile = CSVCreate(real_csv);   // creating the CSV file
+        doorEvents(sensorUuid, sensorName, startTimeMilli, endTimeMilli, csvFile);  // writing to the CSV file
+    }
+
+    private static void _initialize(String apiKey) throws Exception {
+        /*
+         * API CLIENT
+         */
+        {
+            // creating headers for API calls
+            final List<Header> defaultHeaders = new ArrayList<>();
+            defaultHeaders.add(new BasicHeader("x-auth-scheme", "api-token"));
+            defaultHeaders.add(new BasicHeader("x-auth-apikey", apiKey));
+
+            final ClientBuilder clientBuilder = ClientBuilder.newBuilder();
+            clientBuilder.register(new JacksonJaxbJsonProvider().configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false).configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false));
+
+            final HostnameVerifier hostnameVerifier = new HostnameVerifier() {
+                @Override
+                public boolean verify(String hostname, SSLSession session) {
+                    return true;
+                }
+            };
+
+            clientBuilder.hostnameVerifier(hostnameVerifier);
+
+            _apiClient = new ApiClient();
+            _apiClient.setHttpClient(clientBuilder.build());
+            _apiClient.addDefaultHeader("x-auth-scheme", "api-token");
+            _apiClient.addDefaultHeader("x-auth-apikey", apiKey);
+
+            _doorWebservice = new DoorWebserviceApi(_apiClient);
+            _locationWebservice = new LocationWebserviceApi(_apiClient);
+        }
+    }
+
+    // method converts sensor name to uuid
+    public static String uuidConvert(String sensorName) throws Exception {
```

Comment:
all of these functions should be `private static` so that they are not accessible outside of the class